torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -481,24 +482,25 @@ def _root_pre_forward(
     )
     _clear_grads_if_needed(_all_handles(state))
 
+    # Prepares the forward inputs by moving them to ``device``
+    # TODO: Do not use the side stream for tensor copies for now; investigate
+    # the perf with/without it.
+    args_tuple, kwargs_tuple = _to_kwargs(args, kwargs, state.compute_device.index, False)
+    args = args_tuple[0]
```

Comment:
Trying to understand the fix. Is this because `args_tuple[0]` works for root FSDP but not every submodule? Curious to learn what model impl breaks this. I guess it will become clear once we have test for it.

Reply:
sure, will add a unit test once I have a better understanding of the model impl,  this PR does not fix all the cases yet